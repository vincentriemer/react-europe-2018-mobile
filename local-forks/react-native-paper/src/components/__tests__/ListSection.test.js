/* @flow */

import * as React from 'react';
import renderer from 'react-test-renderer';
import ListSection from '../ListSection';
import ListItem from '../ListItem';

it('renders list section without title', () => {
  const tree = renderer
    .create(
      <ListSection>
        <ListItem title="First Item" icon="folder" />
        <ListItem title="Second Item" icon="folder" />
      </ListSection>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list section with title', () => {
  const tree = renderer
    .create(
      <ListSection title="Some title">
        <ListItem title="First Item" icon="folder" />
        <ListItem title="Second Item" icon="folder" />
      </ListSection>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
