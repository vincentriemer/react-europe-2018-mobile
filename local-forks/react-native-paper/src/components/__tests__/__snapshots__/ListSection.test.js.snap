// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list section with title 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 8,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Helvetica Neue",
        },
        Array [
          Object {
            "marginHorizontal": 16,
            "marginVertical": 13,
          },
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "HelveticaNeue-Medium",
          },
        ],
      ]
    }
    theme={
      Object {
        "colors": Object {
          "accent": "#ff4081",
          "background": "#fafafa",
          "disabled": "rgba(0, 0, 0, 0.26)",
          "paper": "#ffffff",
          "placeholder": "rgba(0, 0, 0, 0.38)",
          "primary": "#3f51b5",
          "text": "#000000",
        },
        "dark": false,
        "fonts": Object {
          "light": "HelveticaNeue-Light",
          "medium": "HelveticaNeue-Medium",
          "regular": "Helvetica Neue",
          "thin": "HelveticaNeue-Thin",
        },
        "roundness": 2,
      }
    }
  >
    Some title
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "padding": 8,
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
              ],
            ]
          }
          theme={
            Object {
              "colors": Object {
                "accent": "#ff4081",
                "background": "#fafafa",
                "disabled": "rgba(0, 0, 0, 0.26)",
                "paper": "#ffffff",
                "placeholder": "rgba(0, 0, 0, 0.38)",
                "primary": "#3f51b5",
                "text": "#000000",
              },
              "dark": false,
              "fonts": Object {
                "light": "HelveticaNeue-Light",
                "medium": "HelveticaNeue-Medium",
                "regular": "Helvetica Neue",
                "thin": "HelveticaNeue-Thin",
              },
              "roundness": 2,
            }
          }
        >
          First Item
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "padding": 8,
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
              ],
            ]
          }
          theme={
            Object {
              "colors": Object {
                "accent": "#ff4081",
                "background": "#fafafa",
                "disabled": "rgba(0, 0, 0, 0.26)",
                "paper": "#ffffff",
                "placeholder": "rgba(0, 0, 0, 0.38)",
                "primary": "#3f51b5",
                "text": "#000000",
              },
              "dark": false,
              "fonts": Object {
                "light": "HelveticaNeue-Light",
                "medium": "HelveticaNeue-Medium",
                "regular": "Helvetica Neue",
                "thin": "HelveticaNeue-Thin",
              },
              "roundness": 2,
            }
          }
        >
          Second Item
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders list section without title 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 8,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "padding": 8,
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
              ],
            ]
          }
          theme={
            Object {
              "colors": Object {
                "accent": "#ff4081",
                "background": "#fafafa",
                "disabled": "rgba(0, 0, 0, 0.26)",
                "paper": "#ffffff",
                "placeholder": "rgba(0, 0, 0, 0.38)",
                "primary": "#3f51b5",
                "text": "#000000",
              },
              "dark": false,
              "fonts": Object {
                "light": "HelveticaNeue-Light",
                "medium": "HelveticaNeue-Medium",
                "regular": "Helvetica Neue",
                "thin": "HelveticaNeue-Thin",
              },
              "roundness": 2,
            }
          }
        >
          First Item
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "padding": 8,
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "Helvetica Neue",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
              ],
            ]
          }
          theme={
            Object {
              "colors": Object {
                "accent": "#ff4081",
                "background": "#fafafa",
                "disabled": "rgba(0, 0, 0, 0.26)",
                "paper": "#ffffff",
                "placeholder": "rgba(0, 0, 0, 0.38)",
                "primary": "#3f51b5",
                "text": "#000000",
              },
              "dark": false,
              "fonts": Object {
                "light": "HelveticaNeue-Light",
                "medium": "HelveticaNeue-Medium",
                "regular": "Helvetica Neue",
                "thin": "HelveticaNeue-Thin",
              },
              "roundness": 2,
            }
          }
        >
          Second Item
        </Text>
      </View>
    </View>
  </View>
</View>
`;
